### DISCUSSION TITLE
Is there a way to define the poll frequency when building a consumer ?

### DISCUSSION THREAD

**Discussion by cyrilgeorge-oracle** [confluentinc/parallel-consumer#808] [Q&A] [answered]
Wanted to check if it is possible to configure the poll frequency

**Comment by rkolesnev** [comment, marked as answer]
No, Parallel Consumer has long poll timeout hardcoded at relatively high 2 seconds timeout during normal "Running" mode.
It is a blocking poll - so will return either on getting enough data to at least meet the fetch.min.bytes config or after elapsing.

Note that it is not polling continuously if it is able to fetch messages faster than it can process them - back-pressure kicks in and the wrapped Consumer is paused then until the message buffer is drained below threshold - then it goes back to fetching.

**Reply by cyrilgeorge-oracle** [reply]
Found a way to incorporate poll frequency using pause and resume features. Our durations are in minutes, so the below method is called from another schedule which is invoked with a fixed delay (from last execution end) 2 seconds.
```
 public ScheduledFuture<?> pause(Duration duration) {

        streamProcessor.pauseIfRunning();
        return scheduler.schedule(streamProcessor::resumeIfPaused, duration.toMillis(), TimeUnit.MILLISECONDS);
    }
```
